src/binary/uint.rs

```diff
@@ -73,6 +69,45 @@ impl UInt {
     }
 }
 
+/// The result of the [`magnitude_to_bytes`] function. The fields are private;
+/// users must go through `as_ref()` to see the slice that ignores leading
+/// zeros.
+///
+/// This type exists to avoid heap allocation.
+pub struct U64Magnitude {
+    magnitude_bytes: [u8; mem::size_of::<u64>()],
+    first_occupied_byte: usize,
+}
+
+impl AsRef<[u8]> for U64Magnitude {
+    fn as_ref(&self) -> &[u8] {
+        &self.magnitude_bytes[self.first_occupied_byte..]
+    }
+}
+
+/// Returns the magnitude as big-endian bytes.
+///
+/// ```
+/// use ion_rs::binary::uint;
+/// use std::convert::AsRef;
+///
+/// let repr = uint::magnitude_to_bytes(5u64);
+/// assert_eq!(&[0x05], repr.as_ref());
+/// ```
+pub fn magnitude_to_bytes(magnitude: u64) -> U64Magnitude {
+    // We can divide the number of leading zero bits by 8
+    // to to get the number of leading zero bytes.
+    let empty_leading_bytes: u32 = magnitude.leading_zeros() >> 3;
```

Comment:
*Minor*:  the comment talks about divide by 8, and then you left shift 3 which I grant is the same thing, but if `rustc` can't figure that out on its own from `zeros() / 8`, I would be floored (no math pun `int`ended).  :rofl: *I'll see myself out...*